File: src/fields/Address.ts
import { FieldType } from './index'
import FieldBase, { IFieldBaseDefinition } from './Base'

export interface IFieldAddressValue {
	street1: string
	street2?: string
	city: string
	province: string
	country: string
	zip: string
}

export interface IFieldAddressDefinition extends IFieldBaseDefinition {
	type: FieldType.Address
	value?: IFieldAddressValue
	defaultValue?: IFieldAddressValue
	options?: {}
}

export default class FieldAddress extends FieldBase<IFieldAddressDefinition> {
	public static templateDetails() {
		return {
			definitionInterface: 'IFieldAddressDefinition',
			valueType: 'IFieldAddressValue'
		}
	}
}

File: pkg/fields/address.go
package fields

type AddressFieldValue struct {
	Street1  string `json:"street1"`
	Street2  string `json:"street2,omitempty"`
	City     string `json:"city"`
	Province string `json:"province"`
	Country  string `json:"country"`
	ZIP      string `json:"zip"`
}
